### PR TITLE
Make USM pools optional with a device query to report support.

### PR DESCRIPTION
**Pull request by aarongreig**

### CODE REVIEW

File: test/conformance/usm/usm_adapter_opencl.match [hunk 1]
```diff
@@ -1,86 +1,2 @@
...
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_16_8
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_16_512
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_16_2048
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_32_8
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_32_512
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_32_2048
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_64_8
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_64_512
-urUSMDeviceAllocAlignmentTest.SuccessAlignedAllocations/Intel_R__OpenCL___{{.*}}___UsePoolEnabled_64_2048
 urUSMGetMemAllocInfoTest.Success/Intel_R__OpenCL___{{.*}}___UR_USM_ALLOC_INFO_POOL
```
Comment (aarongreig) [inline test/conformance/usm/usm_adapter_opencl.match:2]: I'm making this an [optional-query] as part of #1332

**Comment by aarongreig** [comment]
LLVM PR https://github.com/intel/llvm/pull/15599

**Review by PietroGhg — APPROVED** [review]
Native CPU changes LGTM, do we also want to change the wording in the `usm*alloc` functions from `"This function must support memory pooling."` to `"This function may support memory pooling."` or something like that?

**Comment by aarongreig** [comment]
> Native CPU changes LGTM, do we also want to change the wording in the `usm*alloc` functions from `"This function must support memory pooling."` to `"This function may support memory pooling."` or something like that?

oh good catch, I'm just going to delete those lines - I think the `UNSUPPORTED_FEATURE` return + explanation are sufficient to imply pooling is optional

**Comment by aarongreig** [comment]
ping @oneapi-src/unified-runtime-maintain @oneapi-src/unified-runtime-opencl-write @oneapi-src/unified-runtime-cuda-write @oneapi-src/unified-runtime-hip-write @oneapi-src/unified-runtime-level-zero-write

**Review by kbenzie — APPROVED** [review]

**Comment by aarongreig** [comment]
@oneapi-src/unified-runtime-cuda-write @oneapi-src/unified-runtime-hip-write @oneapi-src/unified-runtime-level-zero-write please take a look, the adapter changes are pretty trivial

**Review by hdelan — APPROVED** [review]

**Review by nrspruit — APPROVED** [review]
LGTM for L0

**Pull request merged**